### PR TITLE
Provision Managed Network

### PR DESCRIPTION
**Pull request by joshharrin**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.
[Work item to add tests later](https://dev.azure.com/msdata/Vienna/_workitems/edit/2300694/)

### CODE REVIEW

**Review by kwucode — APPROVED** [review]

File: sdk/ml/azure-ai-ml/swagger/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-12-01-preview/machineLearningServices.json [hunk 1]
```diff
@@ -2259,6 +2259,53 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/managedNetwork/provisions": {
```
Comment (seanyao1) [inline sdk/ml/azure-ai-ml/swagger/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-12-01-preview/machineLearningServices.json:2263]: this path doesn't make senses for me. Please work with Kevin to correct this one.

File: sdk/ml/azure-ai-ml/swagger/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-12-01-preview/machineLearningServices.json [hunk 2]
```diff
@@ -2259,6 +2259,53 @@
...
+      "post": {
+        "tags": [
+          "ManagedNetwork"
+        ],
```
Comment (seanyao1) [inline sdk/ml/azure-ai-ml/swagger/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-12-01-preview/machineLearningServices.json:2267]: Why to add this tag on the swagger?

File: sdk/ml/azure-ai-ml/swagger/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-12-01-preview/machineLearningServices.json
Comment (seanyao1) [inline sdk/ml/azure-ai-ml/swagger/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-12-01-preview/machineLearningServices.json:1]: where's this file?

**Review by seanyao1 — CHANGES_REQUESTED** [review]

File: sdk/ml/azure-ai-ml/swagger/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-12-01-preview/machineLearningServices.json (hunk 1 above)
Comment (joshharrin) [reply sdk/ml/azure-ai-ml/swagger/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-12-01-preview/machineLearningServices.json]: Improved path to be {workspaceName}/provisionManagedNetwork

File: sdk/ml/azure-ai-ml/swagger/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-12-01-preview/machineLearningServices.json (hunk 2 above)
Comment (joshharrin) [reply sdk/ml/azure-ai-ml/swagger/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-12-01-preview/machineLearningServices.json]: removed tag from swagger here

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations.py [hunk 3]
```diff
@@ -117,6 +121,27 @@ def begin_sync_keys(self, name: Optional[str] = None) -> LROPoller:
         workspace_name = self._check_workspace_name(name)
         return self._operation.begin_resync_keys(self._resource_group_name, workspace_name)
 
+    @experimental
+    @monitor_with_activity(logger, "Workspace.BeginProvisionNetwork", ActivityType.PUBLICAPI)
+    @distributed_trace
+    def begin_provision_network(
+        self,
+        workspace_name: Optional[str] = None,
+        spark_enabled: Optional[bool] = False,
```
Comment (seanyao1) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations.py:130]: can we make this parameter naming consistent with REST API?

**Review by seanyao1 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations.py (hunk 3 above)
Comment (joshharrin) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations.py]: done, updated to include_spark

**Pull request closed without merging**